anima/env/mayaEnv/publish.py

```diff
@@ -1964,145 +2265,197 @@ def check_sequence_name_format(progress_controller=None):
     if progress_controller is None:
         progress_controller = ProgressControllerBase()
 
+    progress_controller.maximum = 2
+
     # do not consider referenced shot nodes
     shots = pm.ls(type='shot')
-    progress_controller.maximum = len(shots) + 5
     shot = None
     for s in shots:
         if s.referenceFile() is None:
             shot = s
             break
-        progress_controller.increment()
 
     sequencer = shot.outputs(type='sequencer')[0]
-    sequence_name = sequencer.sequence_name.get()
-    progress_controller.increment()
 
-    # SEQ001_003_TNGI
-    # SEQ001_003A_TNGI
-    # EP001_001
+    # get current task
+    from anima.env import mayaEnv
+    m = mayaEnv.Maya()
+    v = m.get_current_version()
+    task = v.task
+
+    # get sequence and scene names
+    scene = None
```

Comment:
oh you are using that code over again. Please wrap this up to a helper function that returns the sequence and scene tasks from the given version and use this function wherever you need this code.

Reply:
is it okay if i keep these helper functions within publish.py or do you recommend somewhere else like utils or something?

Reply:
Yes you can keep them inside ``publish.py``. But if you think that it might be useful outside the publish functionality then you should move them to ``anima.utils``.